### PR TITLE
[New Rule] Kerberos dump using kcc command

### PR DESCRIPTION
**Pull request by Samirbous**
## Issues

Resolves https://github.com/elastic/detection-rules/issues/138

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: rules/macos/credential_access_kerberosdump_kcc.toml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+[metadata]
+creation_date = "2020/08/14"
+ecs_version = ["1.5.0"]
```
Comment (brokensound77) [inline rules/macos/credential_access_kerberosdump_kcc.toml:3]:
```suggestion
ecs_version = ["1.6.0"]
```

File: rules/macos/credential_access_kerberosdump_kcc.toml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+maturity = "production"
+updated_date = "2020/08/14"
+
+[rule]
+author = ["Elastic"]
+description = "Detects the use of kerberos cache manipulation kcc utility to dump cached kerberos tickets."
+index = ["auditbeat-*", "logs-endpoint.events.*"]
+language = "kuery"
+license = "Elastic License"
+name = "Preparation of Keychain crednetials exfiltration"
```
Comment (brokensound77) [inline rules/macos/credential_access_kerberosdump_kcc.toml:13]: looks like a copy pasta from a previous rule

File: rules/macos/credential_access_kerberosdump_kcc.toml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+risk_score = 73
+rule_id = "ad88231f-e2ab-491c-8fc6-64746da26cfe"
+severity = "high"
+tags = ["Elastic", "MacOS"]
+type = "query"
+
+query = '''
+event.category:process and not event.type:end and
+  process.name:kcc and
+  process.args:"copy_cred_cache"
```
Comment (brokensound77) [inline rules/macos/credential_access_kerberosdump_kcc.toml:27]: nit:

```suggestion
  process.args:copy_cred_cache
```

File: rules/macos/credential_access_kerberosdump_kcc.toml (hunk 2 above)
Comment (Samirbous) [reply rules/macos/credential_access_kerberosdump_kcc.toml]: indeed, changed to Kerberos Cached Crednetials Dumping

File: rules/macos/credential_access_kerberosdump_kcc.toml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+maturity = "production"
+updated_date = "2020/08/14"
+
+[rule]
+author = ["Elastic"]
+description = "Detects the use of kerberos cache manipulation kcc utility to dump cached kerberos tickets."
+index = ["auditbeat-*", "logs-endpoint.events.*"]
+language = "kuery"
+license = "Elastic License"
+name = "Kerberos Cached Crednetials Dumping"
```
Comment (brokensound77) [inline rules/macos/credential_access_kerberosdump_kcc.toml:13]:
```suggestion
name = "Kerberos Cached Credentials Dumping"
```

File: rules/macos/credential_access_kerberosdump_kcc.toml [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    "https://opensource.apple.com/source/Heimdal/Heimdal-323.12/kuser/kcc-commands.in.auto.html",
+]
+risk_score = 73
+rule_id = "ad88231f-e2ab-491c-8fc6-64746da26cfe"
+severity = "high"
+tags = ["Elastic", "MacOS"]
+type = "query"
+
+query = '''
+event.category:process and not event.type:end and
```
Comment (brokensound77) [inline rules/macos/credential_access_kerberosdump_kcc.toml:25]:
```suggestion
event.category:process and event_type:(start or process_started) and
```

**Review by brokensound77 — APPROVED** [review]
Just the one last suggestion, then LGTM 👍

File: rules/macos/credential_access_kerberosdump_kcc.toml [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+framework = "MITRE ATT&CK"
+[[rule.threat.technique]]
+id = "T1003"
+name = "Credential Dumping"
+reference = "https://attack.mitre.org/techniques/T1003/"
+
+
+[rule.threat.tactic]
+id = "TA0006"
+name = "Credential Access"
```
Comment (brokensound77) [inline rules/macos/credential_access_kerberosdump_kcc.toml:41]:
```suggestion
name = "OS Credential Dumping"
```

File: rules/macos/credential_access_kerberosdump_kcc.toml [hunk 7]
```diff
@@ -0,0 +1,42 @@
+[metadata]
+creation_date = "2020/08/14"
+ecs_version = ["1.6.0"]
+maturity = "production"
+updated_date = "2020/08/14"
+
+[rule]
+author = ["Elastic"]
+description = "Detects the use of kerberos cache manipulation kcc utility to dump cached kerberos tickets."
```
Comment (bm11100) [inline rules/macos/credential_access_kerberosdump_kcc.toml:9]:
```suggestion
description = "Identifies the use of the Kerberos credential cache (kcc) utility to dump locally cached Kerberos tickets."
```

**Review by bm11100 — APPROVED** [review]
🚀

**Pull request merged**